rdr_service/dao/participant_summary_dao.py

```diff
@@ -1435,46 +1438,87 @@ def get_all_consented_participant_ids(cls, session):
         with self.session() as session:
             query = (
                 sqlalchemy.update(ParticipantSummary).values({
-                    ParticipantSummary.isEhrDataAvailable: False
+                    ParticipantSummary.isEhrDataAvailable: False,
+                    ParticipantSummary.isParticipantMediatedEhrDataAvailable: False
                 })
             )
             return session.execute(query)
 
-    @staticmethod
-    def bulk_update_ehr_status_with_session(session, record_list: List[ParticipantEhrFile]):
+    @classmethod
+    def _bulk_update_ehr_fields(cls, session, record_list: List[ParticipantEhrFile], values_to_update) -> int:
+        if not record_list:
+            return 0
+
         query = (
             sqlalchemy.update(ParticipantSummary)
-            .where(ParticipantSummary.participantId == sqlalchemy.bindparam("pid"))
-            .values(
-                {
-                    ParticipantSummary.ehrStatus.name: EhrStatus.PRESENT,
-                    ParticipantSummary.isEhrDataAvailable: True,
-                    ParticipantSummary.ehrUpdateTime: sqlalchemy.bindparam("receipt_time"),
-                    ParticipantSummary.ehrReceiptTime: sqlalchemy.case(
-                        [(ParticipantSummary.ehrReceiptTime.is_(None), sqlalchemy.bindparam("receipt_time"))],
-                        else_=ParticipantSummary.ehrReceiptTime,
-                    ),
-                }
-            )
+            .where(ParticipantSummary.participantId == sqlalchemy.bindparam('pid'))
+            .values(values_to_update)
         )
-        return session.execute(query, [
+        query_result = session.execute(query, [
             {
                 'pid': record.participant_id,
                 'receipt_time': record.receipt_time
             }
             for record in record_list
         ])
+        return query_result.rowcount
+
+    @classmethod
+    def bulk_update_hpo_ehr_status_with_session(cls, session, record_list: List[ParticipantEhrFile]):
+        return cls._bulk_update_ehr_fields(
+            session=session,
+            record_list=record_list,
+            values_to_update={
+                ParticipantSummary.ehrStatus.name: EhrStatus.PRESENT,
+                ParticipantSummary.isEhrDataAvailable: True,
+                ParticipantSummary.ehrUpdateTime: sqlalchemy.bindparam("receipt_time"),
+                ParticipantSummary.ehrReceiptTime: sqlalchemy.case(
+                    [
+                        (ParticipantSummary.ehrReceiptTime.is_(None), sqlalchemy.bindparam("receipt_time"))
+                    ],
+                    else_=ParticipantSummary.ehrReceiptTime,
+                )
+            }
+        )
+
+    @classmethod
+    def bulk_update_mediated_ehr_status_with_session(cls, session, record_list: List[ParticipantEhrFile]):
+        return cls._bulk_update_ehr_fields(
+            session=session,
+            record_list=record_list,
+            values_to_update={
```

Comment:
The `ehrStatus` field will continue to only apply to HPO EHR, and won't be set to PRESENT for mediated EHR?

Reply:
Correct, the original fields are for HPO, and only the mediated fields get set for the mediated EHR.